code/backend/batch/utilities/tools/QuestionAnswerTool.py

```diff
@@ -1,182 +1,113 @@
-                ),
-                AIMessagePromptTemplate.from_template("{answer}"),
-            ]
-        )
-
-        few_shot_prompt = FewShotChatMessagePromptTemplate(
-            example_prompt=example_prompt,
-            examples=examples,
+        ret_docs_json = json.dumps(
+            {"retrieved_documents": retrieved_documents}, separators=(",", ":")
         )
+        question_with_sources = self.config.prompts.answering_user_prompt.replace(
+            "{sources}", ret_docs_json
+        ).replace("{question}", question)
 
-        answering_prompt = ChatPromptTemplate.from_messages(
-            [
-                SystemMessage(content=self.config.prompts.answering_system_prompt),
-                few_shot_prompt,
-                SystemMessage(content=self.env_helper.AZURE_OPENAI_SYSTEM_MESSAGE),
-                MessagesPlaceholder("chat_history"),
-                HumanMessagePromptTemplate.from_template(
-                    self.config.prompts.answering_user_prompt
-                ),
+        if self.config.prompts.use_on_your_data_format:
+            example_question = self.config.prompts.answering_user_prompt.replace(
+                "{sources}", self.config.example.documents
+            ).replace("{question}", self.config.example.user_question)
+            messages = [
+                # {"role": "system", "content": f"{self.config.prompts.answering_system_prompt} {self.env_helper.AZURE_OPENAI_SYSTEM_MESSAGE}"}, # TODO is this second system message needed?
+                {
+                    "role": "system",
+                    "content": self.config.prompts.answering_system_prompt,
+                },
+                {"role": "system", "name": "example_user", "content": example_question},
+                {
+                    "role": "system",
+                    "name": "example_assistant",
+                    "content": self.config.example.answer,
+                },
             ]
-        )
-
-        documents = json.dumps(
-            {
-                "retrieved_documents": [
-                    {f"[doc{i+1}]": {"content": source.page_content}}
-                    for i, source in enumerate(sources)
-                ],
-            },
-            separators=(",", ":"),
-        )
 
-        return answering_prompt, {
-            "sources": documents,
-            "question": question,
-            "chat_history": chat_history,
-        }
-
-    def answer_question(self, question: str, chat_history: list[dict], **kwargs: dict):
-        # Retrieve documents as sources
-        sources = self.vector_store.similarity_search(
-            query=question,
-            k=self.env_helper.AZURE_SEARCH_TOP_K,
-            filters=self.env_helper.AZURE_SEARCH_FILTER,
-        )
+            for message in chat_history:
+                messages.append(
+                    {"role": message["role"], "content": message["content"]}
+                )  # TODO should this be in the else too?
```

Comment:
When adding the new prompt, I didn't want to make any changes to the old prompt. The old prompt wasn't using chat history at all

Reply:
Wow I didn't realise that!